### ISSUE TITLE
BlenderBim - Perspective camera does not print anything in SVG

### ISSUE THREAD

**Issue by Plae-Blue-Dot** [IfcOpenShell/IfcOpenShell#3880]
Using this version:
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/78321676/f0a53704-9701-4e0b-9ef5-c405573da363)

model view:
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/78321676/9791ce0f-5e49-4dd1-b158-87d87616ccd4)

SVG:
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/78321676/645f9db0-0b62-4294-b882-1e4436d0eedd)

Orthographic camera works fine:
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/78321676/0d5bbae1-3272-4296-b57a-c530571a6ece)

Here is the test file:
[General test 2023.10.12.rar.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/12878796/General.test.2023.10.12.rar.txt)

**Comment by Plae-Blue-Dot** [comment]
Ah got it, thanks Moult! 

Got a result using underlay:
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/78321676/52c18109-7c8b-4757-9a5d-cb317c623e5a)


Excellent! Still it would be great if it was also linework in perspective. Should I make perspective linework a separate feature request/issue or could that be a feature request here?

**Comment by Moult** [comment]
Let's just use this issue. Something for @aothms to see what the requirements are for OCC to parse a perspective view :)

**Comment by aothms** [comment]
For the projector it seems we just need a focal length. https://dev.opencascade.org/doc/refman/html/class_h_l_r_algo___projector.html#ae575f6fb0c5f99edf952451791cb0f3b

But according to

https://dev.opencascade.org/content/how-reach-perspective-projection-using-hlralgoprojector

> HLR algo is just not yet adapted to handle perspective projection properly - it is designed for orthographic projection.

And to be honest I never got it to work.

**Comment by Moult** [comment]
Maybe as a fallback we can use Blender freestyle, what do you guys think? But keep in mind all the semantics will be lost.

**Comment by Plae-Blue-Dot** [comment]
I think that's alright

**Comment by aothms** [comment]
Non-semantic perspective drawings seem to me like a sane compromise, that buys us some time focus on getting ortho in good shape and see what happens in OCC.

Would be good to know though maybe from @yorikvanhavre if they ever got perspective HLR to work in FreeCAD.

**Comment by yorikvanhavre** [comment]
> Would be good to know though maybe from @yorikvanhavre if they ever got perspective HLR to work in FreeCAD.

Yes they did! https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/TechDraw/App/GeometryObject.cpp#L153
